module3/lessons/Week2/FeatureTesting.md

```diff
@@ -0,0 +1,327 @@
+</aside>
+
+<section class="call-to-action">
+    <strong>Practice</strong>
+    <p>In the MvcMovie projects, create a test, and the Controller and Views to satisfy this user story:</p>
+    <span style="white-space: pre-line">As a user
+    When I visith "/Home/FunFacts"
+    Then I see "The Bumble Bee Bat is the smallest mammal."</span>
+</section>
+
+## Setting up our Database Environment
+
+1. Install these packages:
+    * Microsoft.EntityFrameworkCore.InMemory
+2. Add a file to your test project called `appsettings.json`
+3. Update that file to contain the following code:
+
+```json
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Data Source=:memory:"
+  }
+}
+
+```
+
+4. In your test project's `Program.cs` file, update the code with this database connection:
+```c#
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+// You need to add one more using statement:
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using MvcMovie.Controllers;
+using MvcMovie.DataAccess;
+
+namespace MvcMovie.FeatureTests
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.ConfigureServices(services =>
+                    {
+                        //THIS IS THE CODE TO ADD:
+                        services.AddDbContext<MvcMovieContext>(options =>
+                            options.UseInMemoryDatabase("TestDatabase"));
+
+                        services.AddControllersWithViews()
+                            .AddApplicationPart(typeof(HomeController).Assembly);
+                    });
+
+                    webBuilder.Configure(app =>
+                    {
+                        app.UseRouting();
+                        app.UseEndpoints(endpoints =>
+                        {
+                            endpoints.MapControllerRoute(
+                                name: "default",
+                                pattern: "{controller=Home}/{action=Index}/{id?}");
+                        });
+                    });
+                });
+    }
+}
+```
+
+<section class="call-to-action">
+    <p>With a Partner - What similarities and differences do you see between this <code>Program.cs</code> and the <code>Program.cs</code> in the MvcMovie project?</p>
+</section>
+
+## Writing an Index test
+
+<section class="note">
+    <p>In this section, we are going to write a test for the following user story:</p>
+    <span style="white-space: pre-line">As a user
+    When I visith the home page
```

Comment:
The formatting here is a bit hard to read. Maybe format these user stories in a code block?

Reply:
I think the formatting looks cleaner when served by jekyll - I'll take another look just to make sure.